### PR TITLE
Add support for stylelint

### PR DESCRIPTION
**Pull request by blackjid**
We worked with @gmq to add support for https://stylelint.io/

It works quite similar to tslint and eslint.
We've rolled this in our infrastructure and it's looking good

this should close https://github.com/houndci/hound/issues/1037

### CODE REVIEW

File: spec/models/linter/stylelint_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+describe Linter::Stylelint do
+  it_behaves_like "a linter" do
+    let(:lintable_files) { %w(foo.scss) }
+    let(:not_lintable_files) { %w(foo.xxx) }
```
Comment (gylaz) [inline spec/models/linter/stylelint_spec.rb]: Does it lint `sass` files? If not, maybe use that as the `not_lintable_files` extension?

File: spec/models/linter/stylelint_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        pull_request_number: build.pull_request_number,
+        patch: commit_file.patch,
+        content: commit_file.content,
+        config: "{}",
+      )
+    end
+  end
+
+  def stub_scss_config(config = {})
+    stubbed_scss_config = double(
```
Comment (gylaz) [inline spec/models/linter/stylelint_spec.rb]: `double` to `instance_double`?

File: spec/models/linter/stylelint_spec.rb (hunk 1 above)
Comment (blackjid) [inline spec/models/linter/stylelint_spec.rb]: It doen't. Maybe with some extra work it could.

File: spec/models/linter/stylelint_spec.rb (hunk 2 above)
Comment (blackjid) [inline spec/models/linter/stylelint_spec.rb]: sounds like a `instance_double` is a best option here

**Comment by gylaz** [comment]
Thanks again @blackjid!

**Comment by blackjid** [comment]
Awesome!
@gmq great job!

**Pull request merged**